src/Mutator/Util/Mutator.php

```diff
@@ -24,7 +24,7 @@ public function __construct(MutatorConfig $config)
         $this->config = $config;
     }
 
-    abstract public function mutate(Node $node);
+    abstract public function mutate(Node $node): \Generator;
```

Comment:
What are you think to type it as `iterable`?

Reply:
Great suggestion! Done. It's really more powerful

> It accepts any array or object implementing the Traversable interface. Both of these types are iterable using foreach and can be used with yield from within a generator.

Reply:
The downside here is that returning an array is also okay.
And the php-parser api allows to return an array of nodes to replace a set of nodes. Which may cause confusion.

Reply:
booo :(

you are right, returning an array has special meaning and will definitely lead to an error.

https://github.com/nikic/PHP-Parser/blob/7f96481c80ecfe0fd8d37506064d3a25ed9cad23/lib/PhpParser/NodeTraverser.php#L212

I think the last commit should be reverted then

Reply:
If someone curious, with generator we still can return an array to replace one node with several nodes, and we have this example https://github.com/infection/infection/blob/c71d6dd1976d44d5fb932e68cb8d0b83fdc020ff/src/Mutator/ReturnValue/FunctionCall.php#L29-L34

Reply:
Updated my comment. We *want* to return an *array* in some cases, see example above. So technically it's ok to 

* return generator
* return an array if we want to replace one node with several nodes

We just need to cover this in documentation

Reply:
Decided to revert.

With `\Generator`, if we want to replace 1 node by 1 node, we do

```php
yield new Node\...;
```

1 node by 1 node with different cases:

```php
yield new Node\...\1;
yield new Node\...\2;
```

1 node by many new nodes:

```php
yield [new Node\...\1, new Node\...\2];
```

With `iterable`, if we want to replace 1 node by 1 node, we do

```php
return  [new Node\...];
```

1 node by 1 node with different cases:

```php
return [new Node\...\1, new Node\...\2];
```

1 node by many new nodes:

```php
return [[new Node\...\1, new Node\...\2]]; // confusing
```

And with `iterable` we will also able to use `yield`. I agree, this more confusing than powerful.